addons/account/models/account_payment.py

```diff
@@ -484,21 +485,28 @@ def _compute_is_internal_transfer(self):
                                            and payment.partner_id == payment.journal_id.company_id.partner_id \
                                            and payment.destination_journal_id
 
-    @api.depends('available_payment_method_line_ids')
+    @api.depends('available_payment_method_line_ids', 'partner_id')
     def _compute_payment_method_line_id(self):
         ''' Compute the 'payment_method_line_id' field.
         This field is not computed in '_compute_payment_method_line_fields' because it's a stored editable one.
         '''
         for pay in self:
-            available_payment_method_lines = pay.available_payment_method_line_ids
-
-            # Select the first available one by default.
-            if pay.payment_method_line_id in available_payment_method_lines:
-                pay.payment_method_line_id = pay.payment_method_line_id
-            elif available_payment_method_lines:
-                pay.payment_method_line_id = available_payment_method_lines[0]._origin
+            inbound_payment_method = pay.partner_id.property_inbound_payment_method_line_id
+            outbound_payment_method = pay.partner_id.property_outbound_payment_method_line_id
+            if pay.payment_type == 'inbound' and inbound_payment_method:
+                pay.payment_method_line_id = inbound_payment_method
+            elif pay.payment_type == 'outbound' and outbound_payment_method:
+                pay.payment_method_line_id = outbound_payment_method
             else:
-                pay.payment_method_line_id = False
+                available_payment_method_lines = pay.available_payment_method_line_ids
+
+                # Select the first available one by default.
+                if pay.payment_method_line_id.id in available_payment_method_lines.ids:
+                    pay.payment_method_line_id = pay.payment_method_line_id
+                elif available_payment_method_lines:
+                    pay.payment_method_line_id = available_payment_method_lines[0]._origin
+                else:
+                    pay.payment_method_line_id = False
```

Comment:
```py
@api.depends('available_payment_method_line_ids')
def _compute_payment_method_line_id(self):
    ''' Compute the 'payment_method_line_id' field.
    This field is not computed in '_compute_payment_method_line_fields' because it's a stored editable one.
    '''
    for pay in self:
        available_payment_method_lines = pay.available_payment_method_line_ids
        inbound_payment_method = pay.partner_id.property_inbound_payment_method_line_id
        outbound_payment_method = pay.partner_id.property_outbound_payment_method_line_id
        if pay.payment_type == 'inbound' and inbound_payment_method in available_payment_method_lines:
            pay.payment_method_line_id = inbound_payment_method
        elif pay.payment_type == 'outbound' and outbound_payment_method in available_payment_method_lines:
            pay.payment_method_line_id = outbound_payment_method
        elif pay.payment_method_line_id in available_payment_method_lines:
            pay.payment_method_line_id = pay.payment_method_line_id
        elif available_payment_method_lines:
            pay.payment_method_line_id = available_payment_method_lines[0]._origin
        else:
            pay.payment_method_line_id = False
```